### PR TITLE
add missing permissions on host-operator service account

### PR DESCRIPTION
**Pull request by xcoulon**
fixes CRT-301

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
I'm not sure how we can test these changes (besides checking the logs once the operator is deployed), so any suggestion is welcome

**Comment by xcoulon** [comment]
@dipak-pawar @alexeykazakov I re-generated the CSV in https://github.com/codeready-toolchain/host-operator/pull/89/commits/c8c384d88e37522f95f2fbe7d947d03b49e8f10f

File: deploy/role.yaml [hunk 1]
```diff
@@ -15,6 +15,27 @@ rules:
...
+  - processedtemplates
+  - replicationcontrollers
+  - limitranges
+  verbs:
+  - get
+  - list
+- apiGroups: 
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
```
Comment (dipak-pawar) [inline deploy/role.yaml]: @xcoulon, I don't think we need to put it in role, as it's clusterscoped resource, and it's already added in `cluster_role.yaml

File: deploy/role.yaml [hunk 2]
```diff
@@ -15,6 +15,27 @@ rules:
   - secrets
   verbs:
   - "*"
+- apiGroups:
+  - ""
+  resources:
+  - podtemplates
+  - namespaces
+  - componentstatuses
+  - nodes
```
Comment (dipak-pawar) [inline deploy/role.yaml]: As per my understanding, `nodes/namespaces` can be put in `cluster_role.yaml` only, we don't need it to be available in the role

File: deploy/role.yaml (hunk 1 above)
Comment (xcoulon) [reply deploy/role.yaml]: good catch, indeed. I missed the `cluster_role.yaml` file. So I guess the problem we've seen is due to something else, then

Comment (xcoulon) [reply deploy/role.yaml]: so let me revert the last 2 commits

File: deploy/role.yaml (hunk 2 above)
Comment (alexeykazakov) [reply deploy/role.yaml]: Also wondering why would we need nodes even for reading/listing in cluster role... But maybe there are some indirect dependencies?

Comment (MatousJobanek) [reply deploy/role.yaml]: We should need only list permission for nodes - it's needed by KubeFed controller that checks the health of the clusters.
This is already part of the `cluster_role.yaml` file, so no need for additional changes

**Pull request merged**